include/alpaka/rand/Philox/PhiloxBaseCommon.hpp

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/rand/Philox/MultiplyAndSplit64to32.hpp>
+#include <alpaka/rand/Philox/PhiloxConstants.hpp>
+
+#include <utility>
+
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace engine
+        {
+            /** Philox algorithm parameters
+             *
+             * @tparam TCounterSize number of elements in the counter
+             * @tparam TWidth width of one counter element (in bits)
+             * @tparam TRounds number of S-box rounds
+             */
+            // TODO: Not sure where to put this
```

Comment:
Well, make a choice and then remove the comment please. I am sure the next person reading this will have even less of a clue.